### PR TITLE
Filename from uri fix

### PR DESCRIPTION
**Pull request by philn**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
👍

File: src/utils.rs [hunk 1]
```diff
@@ -229,4 +229,17 @@ mod tests {
         check_setenv("Test");
         check_setenv("Тест"); // "Test" in Russian
     }
+
+    #[test]
+    fn test_filename_from_uri() {
+        use gstring::GString;
+        use std::path::PathBuf;
+        let uri: GString = "file:///foo/bar.txt".into();
```
Comment (sdroege) [inline src/utils.rs]: Maybe also add a test with a hostname for completeness?

Comment (philn) [reply src/utils.rs]: In which case can hostname be non-null? I tried an http uri but:
`Error { domain: g_convert_error, code: 4, message: "The URI “http://gtk-rs.org/index.html” is not an absolute URI using the “file” scheme" }`

Comment (sdroege) [reply src/utils.rs]: I don't know :) Maybe check the code of the C function?

Comment (philn) [reply src/utils.rs]: Yes the PR is updated now :)

**Comment by sdroege** [comment]
Perfect :)

**Comment by EPashkin** [comment]
Thanks 👍

**Pull request merged**